### PR TITLE
Aliyun OSS backend support

### PR DESCRIPTION
**Pull request by is**
ref #1354 

- [x] concurrent limit by sem.
- [x] add unit test.
- [x] resolve prune problem.
- [x] gofmt error in travis CI build.

### CODE REVIEW

**Comment by nicle-lin** [comment]
kinda cool, i need this feature,thank you

**Review by fd0 — CHANGES_REQUESTED** [review]
Thanks for giving this a try, please accept my apologies for the long delay! Before this PR can be merged, some things need to be done:
 * We need to find a way to get a test account for development and integration tests with Travis, we have a policy that for each backend a test/devel account must be available, see #1354
 * Squash the commits and rebase it onto master, make two commits: One adding the vendored dependencies, the other adding the new code
 * Add an changelog entry file in `changelog/0.8.2`, let me know if you need help with that

The main discussion is happening in #1354

**Comment by fd0** [comment]
The rclone backend is coming along nicely (see #1657) and rclone's author is already working on integrating this backend. It's currently blocked by the service provider not merging a PR. For details, see #1354.

I'm closing this PR for now. Thanks for the contribution!

**Pull request closed without merging**
